language-php.md

```diff
@@ -2,12 +2,70 @@
 
 We should follow best practices accepted by the community.
 As mentioned in _[php the right way][1]_, PHP-FIG, the PHP Framework Interop Group, announced PSR.
+Some of the style guides are inspired by [CodeIgniter Framework](https://www.codeigniter.com/user_guide/general/styleguide.html)
 
 We decided to follow [PSR-2].
 
 [1]: http://www.phptherightway.com/
 [PSR-2]: https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md
 
+---
+
+Quick Links: [SQL Queries](#sql-queries) | [Localized Texts](#localized-texts) | [Debug Codes](#debug-codes) | [Code Sizes](#code-sizes) | [Operators](#operators)
+
+---
+
+## SQL Queries
+
+Although we’re using an ORM, there are cases where complex SQL statements need raw MySQL queries.
+
+When doing raw MySQL query on PHP, all statements MUST follow our current [mysql style guidelines](https://github.com/juwai/style-guide/blob/master/language-sql.md).
+
+```
+$query = $this->db->query('
+    SELECT
+        `foo`,
+        `bar`,
+        `baz`,
+        `foofoo`,
+        `foobar` AS `raboof`,
+        `foobaz`
+    FROM
+        `exp_pre_email_addresses`
+    WHERE
+        `foo` != 'oof'
+        AND baz != 'zab'
+    ORDER BY `foobaz`
+    LIMIT 5 , 100
+');
+```
+
+## Localized Texts
+
+Any text that is output in the user interface should use language variables in your lang file to allow localization. 
+
+**Valid:**
+```
+return 'Invalid Selection';
+```
+
+**Invalid:**
+```
+return $this->lang->line('invalid_selection');
+```
```

Comment:
Looks like the invalid and valid examples don’t match watch is requested (inverted, most likely).